packages/catalog-generator/src/main/java/io/kaoto/camelcatalog/maven/CamelCatalogVersionLoader.java

```diff
@@ -0,0 +1,314 @@
+                return false;
+            }
+        }
+
+        return areCamelKCRDsLoaded;
+    }
+
+    public void loadLocalSchemas() {
+        loadResourcesFromFolderAsString("schemas", localSchemas, ".json");
+    }
+
+    private void configureRepositories(String version) {
+        if (KAOTO_VERSION_MANAGER.repositories.get("central") == null)
+            KAOTO_VERSION_MANAGER.addMavenRepository("central", "https://repo1.maven.org/maven2/");
+
+        if (version.contains("redhat") && KAOTO_VERSION_MANAGER.repositories.get("maven.redhat.ga") == null)
+            KAOTO_VERSION_MANAGER.addMavenRepository("maven.redhat.ga", "https://maven.repository.redhat.com/ga/");
+    }
+
+    private void loadResourcesFromFolderAsString(String resourceFolderName, Map<String, String> filesMap,
+            String fileSuffix) {
+        ClassLoader classLoader = KAOTO_VERSION_MANAGER.getClassLoader();
+
+        try {
+            Iterator<URL> it = classLoader.getResources(resourceFolderName).asIterator();
+
+            while (it.hasNext()) {
+                URL resourceUrl = it.next();
+
+                if ("jar".equals(resourceUrl.getProtocol())) {
+                    JarURLConnection connection = (JarURLConnection) resourceUrl.openConnection();
+                    JarFile jarFile = connection.getJarFile();
+                    Enumeration<JarEntry> entries = jarFile.entries();
+
+                    while (entries.hasMoreElements()) {
+                        JarEntry entry = entries.nextElement();
+                        if (entry.getName().startsWith(connection.getEntryName()) && !entry.isDirectory()
+                                && entry.getName().endsWith(fileSuffix)) {
+
+                            if (verbose) {
+                                LOGGER.log(Level.INFO, "Parsing: " + entry.getName());
+                            }
+
+                            try (InputStream inputStream = jarFile.getInputStream(entry)) {
+                                try (Scanner scanner = new Scanner(inputStream)) {
+                                    scanner.useDelimiter("\\A");
+                                    String filenameWithoutExtension = entry.getName()
+                                            .replace(resourceFolderName + "/", "")
+                                            .replace(fileSuffix, "");
+                                    filesMap.put(filenameWithoutExtension, scanner.hasNext() ? scanner.next() : "");
+                                }
+                            } catch (IOException e) {
+                                LOGGER.log(Level.SEVERE, e.toString(), e);
+                            }
+                        }
+                    }
+                } else if ("file".equals(resourceUrl.getProtocol())) {
+                    try {
+                        Files.walk(Paths.get(resourceUrl.toURI()))
+                                .filter(Files::isRegularFile)
```

Comment:
```suggestion
                    try (var pathWalker = Files.walk(Paths.get(resourceUrl.toURI()))) {
                    	pathWalker.filter(Files::isRegularFile)
```

Use try-with-resources or close this "Stream" in a "finally" clause.